mailutils-2.2 couldn't be built

## Issue description

mailutils build fails on current master.

## output

```
nixos-rebuild switch -I nixpkgs=~/nixpkgs/
[...]
gcc -c -I. -Iproto -DHAVE_CONFIG_H -DFEAT_GUI_GTK  -I/nix/store/plmc39acnh2pjhyman32x2xax53q8q87-gtk+-2.24.29/include/gtk-2.0 -I/nix/store/plmc39acnh2pjhyman32x2xax53q8q87-gtk+-2.24.29/lib/gtk-2.0/include -I/nix/store/18brg8265mgnrzzll63n6rpzl4r89b1k-glib-2.46.2/include/glib-2.0 -I/nix/store/18brg8265mgnrzzll63n6rpzl4r89b1k-glib-2.46.2/lib/glib-2.0/include -I/nix/store/q0lzlcghx5i65h333sd0dzplmzmyxf3y-cairo-1.14.6/include/cairo -I/nix/store/62pnm2a9g72g6hd1z8v92czhjav4x25c-freetype-2.6.2/include/freetype2 -I/nix/store/62pnm2a9g72g6hd1z8v92czhjav4x25c-freetype-2.6.2/include -I/nix/store/62pnm2a9g72g6hd1z8v92czhjav4x25c-freetype-2.6.2/include/freetype2 -I/nix/store/q8lbgkz2c4f66nxylrqp4sq92094rmwp-fontconfig-2.11.1/include -I/nix/store/ypl7ms14w1qlmyxwfadyc6gdvf9wvmnp-expat-2.1.0/include -I/nix/store/62pnm2a9g72g6hd1z8v92czhjav4x25c-freetype-2.6.2/include/freetype2 -I/nix/store/wyc7dij7cavmn1ik1hhw0ma9qis88kkh-pango-1.38.1/include/pango-1.0 -I/nix/store/0kbdm8zygbqfgnkbsh1ngr7n26hfcs57-gdk-pixbuf-2.32.3/include/gdk-pixbuf-2.0 -I/nix/store/zxx7j2hmc4gwifchyvilid9n06d9bmya-atk-2.18.0/include/atk-1.0   -g -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1        -o objects/menu.o menu.c
  GEN    mu_logger.doc
  GEN    mu_port.doc
  GEN    mu_scm.doc
  GEN    mu_util.doc
  GEN    guile-procedures.texi
ERROR: unknown doc attribute: (cname (hash . hash) (id . scm_mu_address_get_personal) (hash . hash))
Makefile:1349: recipe for target 'guile-procedures.texi' failed
make[2]: *** [guile-procedures.texi] Error 1
make[2]: Leaving directory '/tmp/nix-build-mailutils-2.2.drv-0/mailutils-2.2/libmu_scm'
Makefile:1036: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/tmp/nix-build-mailutils-2.2.drv-0/mailutils-2.2'
Makefile:909: recipe for target 'all' failed
make: *** [all] Error 2
builder for ‘/nix/store/11dpn01nnmmhlvac34xar1fz6vcxx0zl-mailutils-2.2.drv’ failed with exit code 2
cannot build derivation ‘/nix/store/9sv3r188n1gz3r5b3fshhvbljkf9k1f2-system-path.drv’: 1 dependencies couldn't be built
building path(s) ‘/nix/store/26f8v3n2x2r5l771ra12cnmn53inz5py-unit-systemd-vconsole-setup.service’
building path(s) ‘/nix/store/ybvbg2yf9w2ib4yz39k86ghqrv47k502-vlock.pam’
cannot build derivation ‘/nix/store/kgnhjisyazxdg631h1y6j30mjisi556a-nixos-system-binary-16.09.git.a2aec04.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/kgnhjisyazxdg631h1y6j30mjisi556a-nixos-system-binary-16.09.git.a2aec04.drv’ failed
```

## Technical details

* System: NixOS 16.03pre76756.885acea (Emu) -> master on commit https://github.com/NixOS/nixpkgs/commit/a2aec04abcd0887f073355b7cb63641fa4b42a51
* Nix version: nix-env (Nix) 1.11.1
* Nixpkgs version: "16.03pre76756.885acea"

I tried applying the patch from the mailing list. It seems to solve the immediate problem but unfortunately I'm hitting additional problems which I don't know how to solve.

https://github.com/dbrock/nixpkgs/commit/fe17d798449f77882d9a5ae30bda8aad37620b20

```
[...]
  GEN    mu_port.doc
  GEN    mu_scm.doc
  GEN    mu_util.doc
  GEN    guile-procedures.texi
make  all-am
make[3]: Entering directory '/tmp/nix-build-mailutils-2.2.drv-0/mailutils-2.2/libmu_scm'
  CC     mu_address.lo
/nix/store/d20f169ryps7ds2qak0r5n1f4hhxr80h-bash-4.3-p42/bin/bash ../libtool --silent --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I.. -I. -I.. -I../include  -I../include -I../mailbox -I../libproto/include -I/nix/store/65x8pj1ys3p03l7npxdicb598zysbspc-guile-1.8.8/include  -pthread  -DSYSCONFDIR=\"/nix/store/wcxx2sgrqa6l8vkp14q4822xv9pgm134-mailutils-2.2/etc\"  -g -O2 -c -o mu_address.lo mu_address.c
mu_address.c: In function 'mu_scm_address_init':
mu_address.c:183:1: error: expected expression before '}' token
 }
 ^
In file included from mu_address.c:182:0:
./mu_address.x:8:2: error: too few arguments to function 'scm_c_define_gsubr'
  scm_c_define_gsubr (s_scm_mu_username_to_email, 
  ^
In file included from /nix/store/65x8pj1ys3p03l7npxdicb598zysbspc-guile-1.8.8/include/libguile.h:52:0,
                 from ../include/mailutils/guile.h:23,
                 from mu_scm.h:53,
                 from mu_address.c:20:
/nix/store/65x8pj1ys3p03l7npxdicb598zysbspc-guile-1.8.8/include/libguile/gsubr.h:47:13: note: declared here
 SCM_API SCM scm_c_define_gsubr (const char *name, 
             ^
mu_address.c:183:1: error: expected ')' before '}' token
 }
 ^
[...]
```

--- a/pkgs/tools/networking/mailutils/default.nix
+++ b/pkgs/tools/networking/mailutils/default.nix
@@ -1,7 +1,6 @@
 { fetchurl, stdenv, gettext, gdbm, libtool, pam, readline
-, ncurses, gnutls, mysql, guile, texinfo, gnum4, dejagnu, sendmailPath ? "/var/setuid-wrappers/sendmail" }:
-
-/* TODO: Add GNU SASL, GNU GSSAPI, and FreeBidi.  */
+, ncurses, gnutls, sasl, fribidi, gss , mysql, guile, texinfo,
+  gnum4, dejagnu, nettools }:
 
 stdenv.mkDerivation rec {
   name = "mailutils-2.2";
@@ -11,17 +10,19 @@ stdenv.mkDerivation rec {
     sha256 = "0szbqa12zqzldqyw97lxqax3ja2adis83i7brdfsxmrfw68iaf65";
   };
 
-  patches = [ ./path-to-cat.patch ./no-gets.patch ];
+  patches = [ ./path-to-cat.patch ./no-gets.patch ./scm_c_string.patch ];
 
-  configureFlags = "--with-path-sendmail=${sendmailPath}";
+  configureFlags = [
+    "--with-gsasl"
+    "--with-gssapi=${gss}"
+  ];
 
   buildInputs =
    [ gettext gdbm libtool pam readline ncurses
-     gnutls mysql.lib guile texinfo gnum4 ]
+     gnutls mysql.lib guile texinfo gnum4 sasl fribidi gss nettools ]
    ++ stdenv.lib.optional doCheck dejagnu;
 
-  # Tests fail since gcc 4.8
-  doCheck = false;
+  doCheck = true;
 
   meta = with stdenv.lib; {
     description = "Rich and powerful protocol-independent mail framework";
@@ -51,7 +52,7 @@ stdenv.mkDerivation rec {
       gpl3Plus /* tools */
     ];
 
-    maintainers = [ ];
+    maintainers = with maintainers; [ vrthra ];
 
     homepage = http://www.gnu.org/software/mailutils/;
 

--- a/pkgs/tools/networking/mailutils/scm_c_string.patch
+++ b/pkgs/tools/networking/mailutils/scm_c_string.patch
@@ -1,0 +1,15 @@
+See https://lists.gnu.org/archive/html/bug-mailutils/2010-10/msg00005.html
+
+diff -u mailutils-2.2/libmu_scm/mu_message.c mailutils-2.2.new/libmu_scm/mu_message.c
+--- mailutils-2.2/libmu_scm/mu_message.c	2010-04-18 10:29:07.000000000 -0700
++++ mailutils-2.2.new/libmu_scm/mu_message.c	2016-07-03 21:18:53.746185547 -0700
+@@ -510,8 +510,8 @@
+     {
+       SCM car = SCM_CAR (lst);
+       if (scm_is_string (car)
+ 	  && mu_c_strncasecmp (scm_i_string_chars (car), name,
+-			       scm_i_string_length (car)) == 0)
++			       scm_c_string_length (car)) == 0)
+ 	return 1;
+     }
+   return 0;

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -2420,7 +2420,7 @@ in
   mailpile = callPackage ../applications/networking/mailreaders/mailpile { };
 
   mailutils = callPackage ../tools/networking/mailutils {
-    guile = guile_1_8;
+    sasl = gsasl;
   };
 
   email = callPackage ../tools/networking/email { };

